extensions/binding/org.eclipse.smarthome.binding.lifx/src/main/java/org/eclipse/smarthome/binding/lifx/handler/LifxLightHandler.java

```diff
@@ -84,158 +235,503 @@ public void handleCommand(ChannelUID channelUID, Command command) {
+                                unicastChannel.configureBlocking(false);
+                                unicastKey = unicastChannel.register(selector,
+                                        SelectionKey.OP_READ | SelectionKey.OP_WRITE);
+                                unicastChannel.connect(ipAddress);
+                                logger.trace("Connected to a bulb via {}", unicastChannel.getLocalAddress().toString());
+                            } catch (Exception e) {
+                                logger.warn("An exception occurred while connectin to the bulb's IP address : '{}'",
+                                        e.getMessage());
+                                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                                return;
+                            }
+
+                            updateStatus(ThingStatus.ONLINE);
+
+                            // populate the current state variables
+                            GetLightPowerRequest powerPacket = new GetLightPowerRequest();
+                            sendPacket(powerPacket);
+
+                            GetRequest colorPacket = new GetRequest();
+                            sendPacket(colorPacket);
+
+                        }
+                    }
+                }
+
+                return;
+            }
+
+            if (packet instanceof StateResponse) {
+                handleLightStatus((StateResponse) packet);
+                return;
+            }
+
+            if (packet instanceof StatePowerResponse) {
+                handlePowerStatus((StatePowerResponse) packet);
+                return;
+            }
+
+            if (packet instanceof StateLabelResponse) {
+                // Do we care about labels?
+                return;
+            }
+
+            if (packet instanceof EchoRequestResponse) {
+                handleEchoRequestReponse((EchoRequestResponse) packet);
+                return;
+            }
+        }
     }
 
-    private LFXHSBKColor toLFXHSBKColor(HSBType color) {
-        LFXHSBKColor lfxColor = LFXHSBKColor.getColor(color.getHue().intValue(),
-                color.getSaturation().floatValue() / 100, color.getBrightness().floatValue() / 100, 0);
-        return lfxColor;
+    public void handleLightStatus(StateResponse packet) {
+        DecimalType hue = new DecimalType(packet.getHue() * 360 / 65535.0f);
+        PercentType saturation = new PercentType(Math.round((packet.getSaturation() / 65535.0f) * 100));
+        PercentType brightness = new PercentType(Math.round((packet.getBrightness() / 65535.0f) * 100));
+        DecimalType temperature = new DecimalType(packet.getKelvin());
+
+        currentColorState = new HSBType(hue, saturation, brightness);
+        currentPowerState = packet.getPower();
+        currentTempState = temperature;
+
+        if (currentPowerState == PowerState.OFF) {
+            updateState(CHANNEL_COLOR, OnOffType.OFF);
```

Comment:
This will set the internal state of the Color item to OFF instead of HSB(x,y,0), so information about hue and saturation is lost. That's why I suggest to change this to
```
updateState(CHANNEL_COLOR, new HSBType(hue, saturation, 0);
```